Rakefile

```diff
@@ -1,30 +1,11 @@
-require 'rdoc/task'
-
-task :default => :test
+# frozen_string_literal: true
 
-# We test this way because of what this library does.
-# The tests wrap and load C++ wrapper code constantly.
-# When running all the tests at once, we very quickly run 
-# into problems where Rice crashes because 
-# a given C++ class is already wrapped, or glibc doesn't like our 
-# unorthodox handling of it's pieces. So we need to run the
-# tests individually
-desc "Run the tests"
-task :test do
-  require 'rbconfig'
-  FileList["test/*_test.rb"].each do |file|
-    # To allow multiple ruby installs (like a multiruby test suite), I need to get
-    # the exact ruby binary that's linked to the ruby running the Rakefile. Just saying
-    # "ruby" will find the system's installed ruby and be worthless
-    ruby = File.join(RbConfig::CONFIG["bindir"], RbConfig::CONFIG["RUBY_INSTALL_NAME"])
-    sh "#{ruby} -Itest -S rspec #{file}"
-  end
-end
```

Comment:
I think this library should still have individual test runs. There are probably improvements to be made here but having a clean Ruby for every test is important to ensure that compiled extensions aren't causing test conflicts.

Reply:
IMO, we could do a 
```ruby
config.before do
   FileUtils.rm_rf(...)
end
```

that way it's fresh

Reply:
That's not what I'm talking about. I mean the Ruby execution process itself. This library builds Ruby extensions, and to test itself it builds a ton of specialized extensions and loads them into memory. You can't unload an extension from memory outside of shutting down the Ruby process itself. That's why we run each test file individually, so it's a clean slate for just the extension logic we're testing.